syncany-lib/src/main/java/org/syncany/operations/up/Indexer.java

```diff
@@ -367,11 +330,21 @@ private void addFileVersion(FileProperties fileProperties) {
 			PartialFileHistory lastFileHistory = guessLastFileHistory(fileProperties);
 			FileVersion lastFileVersion = (lastFileHistory != null) ? lastFileHistory.getLastVersion() : null;
 
-			// 2. Create new file history/version
+			// 2. If file type changed, "close" the old file history by adding a file version that deletes the old file/directory
+			PartialFileHistory deletedFileHistory = null;
```

Comment:
Previously we would do the file deletion by iterating over the entire database (https://github.com/syncany/syncany/blob/develop/syncany-lib/src/main/java/org/syncany/operations/up/Indexer.java#L210). I needed to remove the cache that was used here, and I switched to using the output from Status to do deletion. However, the StatusOperation does not detect type changes and having a file twice in the changeSet (one time as added, one time as deleted) seemed a bit dangerous.